File: backend/src/main/java/com/mynt/mynt/model/CurrencyCloud.java
package com.mynt.mynt.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "\"CurrencyCloud\"")
public class CurrencyCloud {
    @Id
    @Column(name = "user_id", nullable = false)
    private Integer id;

    @MapsId
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "\"UUID\"", nullable = false, length = 100)
    private String uuid;

}

File: backend/src/main/java/com/mynt/mynt/model/CurrencyCloud.kt
package com.mynt.mynt.model

import jakarta.persistence.*

@Entity
@Table(name = "\"CurrencyCloud\"")
open class CurrencyCloud {
    @Id
    @Column(name = "user_id", nullable = false)
    open var id: Int? = null

    @MapsId
    @OneToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    open var user: User? = null

    @Column(name = "\"UUID\"", nullable = false, length = 100)
    open var uuid: String? = null
}